src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioTransformationServiceImpl.java

```diff
@@ -231,6 +234,15 @@ private UUID processAudioRequest(Integer requestId) {
         return blobId;
     }
 
+    List<MediaEntity> filterMediaByMediaRequestDates(List<MediaEntity> mediaEntitiesForRequest, MediaRequestEntity mediaRequestEntity) {
+        return mediaEntitiesForRequest.stream().filter(
+            media -> (media.getStart().isAfter(mediaRequestEntity.getStartTime()) && media.getStart().isBefore(mediaRequestEntity.getEndTime()))
+                || (media.getEnd().isBefore(mediaRequestEntity.getEndTime()) && media.getEnd().isAfter(mediaRequestEntity.getEndTime()))
+                || (media.getStart().isBefore(mediaRequestEntity.getStartTime()) && media.getEnd().isAfter(mediaRequestEntity.getEndTime()))
+                || media.getStart().isEqual(mediaRequestEntity.getStartTime()) || media.getEnd().isEqual(mediaRequestEntity.getEndTime()))
```

Comment:
Is the requirement for the filter to pass any `media` whose timespan overlaps with the `media_request` timespan? If so, I'm wondering if the following would be sufficient to cover the use cases:

```java
    List<MediaEntity> filterMediaByMediaRequestDates(List<MediaEntity> mediaEntitiesForRequest, MediaRequestEntity mediaRequestEntity) {
        return mediaEntitiesForRequest.stream()
            .filter(media ->
                        (mediaRequestEntity.getStartTime()).isBefore(media.getEnd())
                            && (media.getStart().isBefore(mediaRequestEntity.getEndTime())))
            .collect(Collectors.toList());
    }
```

Reply:
I tried it out all bar one test worked. The test that failed was where the end of the first media is included but shouldn't be as its the end of the audio so nothing to trim:

```
List<MediaEntity> mediaEntities = createMediaEntities(TIME_11_59, TIME_12_20, TIME_12_20, TIME_12_40, TIME_12_40, TIME_13_01);
        MediaRequestEntity mediaRequestEntity = createMediaRequest(TIME_12_00, TIME_13_00);
```

The media that ends at 12:20 would be trimmed to 12:20 so that would result in no data. It was one of the reasons I was unsure about the code, besides the fact my code was complicated. I don't yet know if the precision is minutes or seconds.

Reply:
Let's have a chat on this one as my interpretation of the statement on DMP-1615 "The logic should pull in all the Audio files falling within the time range in media request" is a little different. My understanding is:

Given the media request spans from 12:00 to 13:00:
- The first `media` should pass the filter, as it spans from 11:59 - 12:20, so it has 20 mins of wanted audio
- The second `media` should pass the filter, as it spans from 12:20-12:40, so it has 20 mins of wanted audio
- The third `media` should pass the filter, as it spans from 12:40-13:01, so it has 20 mins of wanted audio